src/System.Runtime.InteropServices.RuntimeInformation/src/System/Runtime/InteropServices/RuntimeInformation/RuntimeInformation.cs

```diff
@@ -22,7 +23,10 @@ public static string FrameworkDescription
         {
             get
             {
-                return s_frameworkDescription ?? (s_frameworkDescription = $"{FrameworkName} {typeof(object).GetTypeInfo().Assembly.GetName().Version}");
+                AssemblyFileVersionAttribute attr = (AssemblyFileVersionAttribute)(typeof(object).GetTypeInfo().Assembly.GetCustomAttribute(typeof(AssemblyFileVersionAttribute)));
+                Debug.Assert(attr != null);
+                return s_frameworkDescription ?? 
```

Comment:
`s_frameworkDescription` cache is not effective after this change because of the expensive reflection operation is done unconditionally, even when there is good value cached already.